app/lib/actions/pulp3/repository/multi_copy_content.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Pulp3
+    module Repository
+      class MultiCopyContent < Pulp3::AbstractAsyncTask
+        def plan(extended_repo_map, smart_proxy, options)
+          repo_id_map = {}
+          filter_ids = []
+
+          extended_repo_map.each do |source_repos, dest_repo_map|
+            filter_ids << dest_repo_map[:filters]&.map(&:id)
+            repo_id_map[source_repos&.map(&:id)] = { :dest_repo => dest_repo_map[:dest_repo].id }
+          end
+
+          plan_self(options.merge(:repo_id_map => repo_id_map,
+                                  :smart_proxy_id => smart_proxy.id,
+                                  :filter_ids => filter_ids.flatten))
+        end
+
+        def invoke_external_task
+          repo_id_map = {}
+
+          input[:repo_id_map].each do |source_repo_ids, dest_repo_id|
+            repo_id_map[JSON.parse(source_repo_ids)] = dest_repo_id
```

Comment:
any idea why the JSON.parse here was needed?

Reply:
My repo ID map ended up coming back as a string for some reason so I had to do a JSON.parse.

Reply:
i see that too.  It looks like json doesn't support an array as a hash key, so it converts it to a string